File: ouCapstone/code v2/fib.c
#include <stdio.h>

int fib(n)
{
	if (n <= 2)
		return 1;
	else
		return fib(n-1) + fib(n-2);
}

int main(int argc, char **argv) 
{
	int n;
	char string[80];
	printf("What Fibonacci Number would you like to compute? ");
	scanf("%s",string);
	getchar();
	n = atoi(string);
	printf("fib(%d) = %d\n", n, fib(n));
	printf("Press [ENTER] to close");
	while( getchar() != '\n' );
	return 0;
}

File: ouCapstone/code v2/fib.py
#!/usr/bin/python

import sys, getopt

def main(argv):
   num = ''
   try:
      opts, args = getopt.getopt(sys.argv[1:],"h:n:",["number="])
      if not opts:
         num = int(raw_input("Which fibonacci number do you wish to see? "))
      else:
         numtemp = int(raw_input("Which fibonnacci number do you wish to see? [" + sys.argv[2] + "] ") or sys.argv[2])
   except getopt.GetoptError:
      print 'fib.py -n <number>'
      sys.exit(2)        
   for opt, arg in opts:
      if opt == '-h':
         print 'fib.py -n <number>'
         sys.exit()
      elif opt in ("-n", "--number"):
         num = int(arg)
   print fib(num)

def fib(n):
 a,b = 1,1
 for i in range(n-1):
  a,b = b,a+b
 return a

if __name__ == "__main__":
   main(sys.argv[1:])
   raw_input("Press [RETURN] to continue")